File: HX.AI-Chat/HX.AI-Chat.Dto/ConversationDto.cs
namespace HX.AI_Chat.Dto
{
    public class ConversationDto
    {
        public Guid Id { get; set; }

        public string Name { get; set; } = null!;

        public DateTimeOffset DateCreated { get; set; }

        public DateTimeOffset DateModified { get; set; }
    }

    public class ChatConversationDto : ConversationDto
    {
        public List<ConversationMessageDto> Messages { get; set; } = [];
    }
}

File: ai-chat-ui/src/app/dtos/ConversationDto.ts
export interface ConversationDto {
  id: string;
  name: string;
  projectId?: string;
  dateCreated: string; // ISO date string from API
  dateModified: string; // ISO date string from API
}

export interface ConversationMessageDto {
  id: string;
  name: string;
  dateCreated: string;
  dateModified: string;
  text: string;
  role: number;
}

export interface ChatConversationDto {
  id: string;
  name: string;
  dateCreated: string;
  dateModified: string;
  messages: ConversationMessageDto[];
}
